trace_replay/block_cache_tracer.cc

```diff
@@ -29,23 +41,10 @@ BlockCacheTraceWriter::BlockCacheTraceWriter(
       trace_options_(trace_options),
       trace_writer_(std::move(trace_writer)) {}
 
-bool BlockCacheTraceWriter::ShouldTrace(
-    const BlockCacheTraceRecord& record) const {
-  if (trace_options_.sampling_frequency == 0 ||
-      trace_options_.sampling_frequency == 1) {
-    return true;
-  }
-  // We use spatial downsampling so that we have a complete access history for a
-  // block.
-  const uint64_t hash = GetSliceNPHash64(Slice(record.block_key));
-  return hash % trace_options_.sampling_frequency == 0;
-}
-
 Status BlockCacheTraceWriter::WriteBlockAccess(
     const BlockCacheTraceRecord& record) {
   uint64_t trace_file_size = trace_writer_->GetFileSize();
-  if (trace_file_size > trace_options_.max_trace_file_size ||
-      !ShouldTrace(record)) {
+  if (trace_file_size > trace_options_.max_trace_file_size) {
```

Comment:
Maybe add logging statement here when file size threshold is reached?

Reply:
As discussed offline, I will do this in another PR.